packages/core/src/base/authHelper.ts

```diff
@@ -33,14 +33,16 @@ export function stripBearer(header: string): string {
  * @class ShopperToken
  * @implements {IAuthToken}
  */
-export class ShopperToken implements IAuthToken {
+export class ShopperToken<T> implements IAuthToken {
   public rawToken: string;
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   public decodedToken: { [key: string]: any } | string;
+  public body: T;
 
-  constructor(token: string) {
+  constructor(dto: T, token: string) {
     this.rawToken = token;
     this.decodedToken = decode(this.rawToken);
+    this.body = dto;
```

Comment:
Calling this "body" is a little confusing too me and exposes an implementation detail. It's the body of the response, but not the body of the token, which is what it's a field of. I know we're doing the variable type thing here, but what the user should always expect it to be is customer info, so maybe just call it "customerInfo"?